ffmpeg/lpms_ffmpeg.c

```diff
@@ -272,6 +272,45 @@ int lpms_rtmp2hls(char *listen, char *outf, char *ts_tmpl, char* seg_time, char
   return ret == AVERROR_EOF ? 0 : ret;
 }
 
+//
+// Bypass Check
+// this is needed to handle streams that have first few segments that are
+// audio-only (i.e. have a video stream but no frames)
+// returns: 0 if both audio/video streams valid
+//          1 for video with 0-frame, that needs bypass
+//          <0 invalid stream(s) or internal error
+//
+int lpms_is_bypass_needed(char *fname)
+{
+  AVFormatContext *ic = NULL;
+  int ret = 0, vstream = 0, astream = 0;
+
+  ret = avformat_open_input(&ic, fname, NULL, NULL);
+  if (ret < 0) { ret = -1; goto close_format_context; }
+  ret = avformat_find_stream_info(ic, NULL);
+  if (ret < 0) { ret = -1; goto close_format_context; }
+
+  vstream = av_find_best_stream(ic, AVMEDIA_TYPE_VIDEO, -1, -1, NULL, 0);
+  astream = av_find_best_stream(ic, AVMEDIA_TYPE_AUDIO, -1, -1, NULL, 0);
+  if (vstream >= 0 && astream >= 0) {
+      if (AV_PIX_FMT_NONE == ic->streams[vstream]->codecpar->format &&
+          0 == ic->streams[vstream]->codecpar->height) {
+          // no valid pixel format and picture height => needs bypass
```

Comment:
Just to clarify: we treat AV_PIX_FMT_NONE as a signal that the video stream has 0 frames? Could there ever be a scenario where there is no valid pixel format, but the video stream has > 0 frames?

Reply:
> Could there ever be a scenario where there is no valid pixel format, but the video stream has > 0 frames?
Yep it might happen for some weird source pixel format that FFmpeg doesn't recognize, but anyway the encoders won't recognize it either so it'll fail at a later stage unless the user selected `copy`.

But with the extra check of checking `codecpar->height` ([AVCodecParameters.height](https://ffmpeg.org/doxygen/trunk/structAVCodecParameters.html#a1ec57ee84f19cf65d00eaa4d2a2253ce)) here is a reliable indicator of having 0-frames, even more so than `streavs[vstream]->nb_frames` ([AVStream.nb_frames](https://ffmpeg.org/doxygen/trunk/structAVStream.html#a4382c3064df1c9eb232ac198dec067f9)) which is not always set by default.